src/test/java/emissary/core/sentinel/protocols/rules/AllMaxTimeTest.java

```diff
@@ -51,4 +60,54 @@ void notOverTimeLimit() {
         Rule rule = new AllMaxTime("rule1", "TestPlace", 2, 0.75);
         assertFalse(rule.overTimeLimit(placeAgentStats));
     }
+
+    @Test
+    void condition() {
```

Comment:
(nit) This method's got a lot of redundant code.  Can make it somewhat less dense with a couple minor changes:
- create a `stats` variable to reuse for all the condition checks
```
final List<PlaceAgentStats> stats = List.of(lowerStats, upperStats);
```
- eliminate consecutive `when(pool.getCurrentPoolSize()).thenReturn(value);` calls that don't change the existing pool size. The noise obscures the intent of the assertions.   Might also be clarifying to wrap those in a `setPoollSize(pool, newSize)` helper method.